.eslintrc.json

```diff
@@ -28,25 +37,37 @@
             "style": "kebab-case"
           }
         ],
+        "@typescript-eslint/ban-types": "warn",
+        "@typescript-eslint/no-empty-function": "warn",
+        "@typescript-eslint/no-inferrable-types": "warn",
+        "@typescript-eslint/no-this-alias": "warn",
+        "no-dupe-class-members": "warn",
+        "no-prototype-builtins": "warn",
+        "no-unused-vars": "off",
+        "no-useless-escape": "warn",
+        "no-var": "warn",
         "quotes": [
           "warn",
           "single",
           {
             "allowTemplateLiterals": true
           }
-        ]
+        ],
+        "prefer-const": "warn",
+        "prettier/prettier": "warn"
```

Comment:
I have marked these issues as a warning, for now, to keep this PR small and consumable. It would be great to raise PRs to address each of these issues later.